File: src/entities/SeriesRating.ts
import { Column, Entity, JoinColumn, ManyToOne } from "typeorm";
import { Series } from "./Series";
import { Users } from "./Users";

@Entity("series_rating", { schema: "dionysus" })
export class SeriesRating {
  @Column("integer", { name: "rating", nullable: true })
  rating: number | null;

  @ManyToOne(() => Series, (series) => series.seriesRatings, {
    onDelete: "CASCADE",
    onUpdate: "CASCADE",
  })
  @JoinColumn([{ name: "series_id", referencedColumnName: "seriesId" }])
  series: Series;

  @ManyToOne(() => Users, (users) => users.seriesRatings, {
    onDelete: "CASCADE",
    onUpdate: "CASCADE",
  })
  @JoinColumn([{ name: "user_id", referencedColumnName: "userId" }])
  user: Users;
}

File: engine/engine.api/Models/SeriesRating.cs
using System;
using System.Collections.Generic;

namespace engine.api.Models
{
    public partial class SeriesRating
    {
        public int? Rating { get; set; }
        public long? UserId { get; set; }
        public long? SeriesId { get; set; }

        public virtual Series? Series { get; set; }
        public virtual User? User { get; set; }
    }
}
